src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -1634,6 +1689,74 @@ def __ne__(self, other):
         """
         return not self.__eq__(other)
 
+    def __le__(self, other):
+        """
+        Compare orders self and other.
```

Comment:
```suggestion
        Compare orders ``self`` and ``other``.
```
You should also use a `__richcmp__` instead. (If not, I forget if Python also requires all of the others to be implemented if you have just a, e.g., `__lt__`.)

Reply:
Regarding `__richcmp__`: I tried implementing that, but it resulted in multiple errors in testing; I'm not sure why, though.

Reply:
The following is how I tried to implement it:
```
def __richcmp__(self, right, op):
    return self.unit_ideal()._richcmp_(right.unit_ideal(), op)
```
However, building and testing with this gives me the error 
`TypeError: '<=' not supported between instances of 'QuaternionOrder_with_category' and 'QuaternionOrder_with_category'`

Maybe I'm misunderstanding how `__richcmp__` is supposed to be implemented though.

Reply:
You would  use
```python
def __richcmp__(self, other, op):
    from sage.structure.richcmp import richcmp
    return richcmp(self.unit_ideal(), other.unit_ideal(), op)
```

Reply:
Also you mark the class as `@richcmp_method` from `from sage.structure.richcmp importrichcmp_method`.

Reply:
Thanks, that worked! Does the `_richcmp_`-method of the class `QuaternionFractionalIdeal_rational` need similar changes?

Reply:
I don't know. I'm not familiar with that code. So you will have to test, but it might be okay.

Reply:
It seems to work fine, so I'll leave it as is - I'm just surprised that it does work, given the extra steps that were necessary for the `__richcmp__` here.